src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,158 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        private Node(Node<T> prev, T value, Node<T> next) {
+            this.value = value;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        addTail(value);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Wrong index!");
+        }
+        if (index == size) {
+            addTail(value);
+        } else {
+            addHead(value, getNode(index));
+        }
+    }
+
+    private void addTail(T value) {
+        Node<T> lastNode = tail;
+        Node<T> node = new Node<>(lastNode, value, null);
+        tail = node;
+        if (lastNode == null) {
+            head = node;
+        } else {
+            lastNode.next = node;
+        }
+        size++;
+    }
 
+    private void addHead(T value, Node<T> node) {
+        Node<T> nodeBefore = node.prev;
+        Node<T> newNode = new Node<>(nodeBefore, value, node);
+        node.prev = newNode;
+        if (nodeBefore == null) {
+            head = newNode;
+        } else {
+            nodeBefore.next = newNode;
+        }
+        size++;
+    }
+
+    private Node<T> getNode(int index) {
+        checkSize(index);
+        Node<T> nodeFromHead = head;
+        Node<T> nodeFromTail = tail;
+        if (index <= size / 2) {
+            for (int i = 0; i < index; i++) {
+                nodeFromHead = nodeFromHead.next;
+            }
+            return nodeFromHead;
+        } else {
+            for (int i = 0; i < (size - 1) - index; i++) {
+                nodeFromTail = nodeFromTail.prev;
+            }
+            return nodeFromTail;
+        }
+    }
+
+    private void unlink(Node<T> node) {
+        if (node == tail) {
```

Comment:
```suggestion
        if (node.next == null) {
```
it's a bad practice to compare objects with ==. 
This way you can know if node is tail